pyccl/background.py

```diff
@@ -112,6 +113,25 @@ def comoving_angular_distance(cosmo, a):
                          lib.comoving_angular_distance_vec, cosmo, a)
 
 
+def comoving_angular_diameter_distance(cosmo, a1, a2):
+    """Comoving angular diameter distance.
+
+   .. note:: The comovoing angular diameter distance in Mpc from scale factor
+    a1 to scale factor a2. This is Eq. (19) of astro-ph/9905116.
+
+    Args:
+        cosmo (:obj:`Cosmology`): Cosmological parameters.
+        a1 (float or array_like): Scale factor(s), normalized to 1 today.
+        a2 (float or array_like): Scale factor(s), normalized to 1 today.
+
+    Returns:
+        float or array_like: Comoving angular distance; Mpc.
+    """
```

Comment:
You need to make sure the spline data is initialized by calling `cosmo.compute_distances()`.

Reply:
Done